gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcSharedPolicyGroupRepository.java

```diff
@@ -91,4 +103,51 @@ public Optional<SharedPolicyGroup> findById(String id) throws TechnicalException
     public Set<SharedPolicyGroup> findAll() throws TechnicalException {
         throw new IllegalStateException("Not implemented");
     }
+
+    @Override
+    public Page<SharedPolicyGroup> search(SharedPolicyGroupCriteria criteria, Pageable pageable, Sortable sortable)
+        throws TechnicalException {
+        Objects.requireNonNull(pageable, "Pageable must not be null");
+        Objects.requireNonNull(criteria, "SharedPolicyGroupCriteria must not be null");
+        Objects.requireNonNull(criteria.getEnvironmentId(), "EnvironmentId must not be null");
+        LOGGER.debug("JdbcSharedPolicyGroupRepository.search({}, {})", criteria.toString(), pageable.toString());
+
+        try {
+            final var name = criteria.getName() == null ? "%" : "%" + criteria.getName() + "%";
+
+            var total = jdbcTemplate.queryForObject(
+                "select count(*) from " + this.tableName + " WHERE environment_id = ? AND lower(name) like ?",
+                Long.class,
+                criteria.getEnvironmentId(),
+                name
+            );
+
+            if (total == null || total == 0) {
+                return new Page<>(List.of(), pageable.pageNumber(), 0, 0);
+            }
+
+            sortable = sortable == null ? new SortableBuilder().field("created_at").setAsc(true).build() : sortable;
+            final var sortOrder = sortable.order() == Order.ASC ? "ASC" : "DESC";
+            final var sortField = toSnakeCase(sortable.field());
+
+            var result = jdbcTemplate.query(
+                getOrm().getSelectAllSql() +
+                " WHERE environment_id = ? AND lower(name) like ? ORDER BY" +
+                " " +
+                sortField +
+                " " +
+                sortOrder +
+                " " +
```

Comment:
```suggestion
                ? +
                " " +
                ? +
                " " +
```

you should prefer using parameterized query to avoid sql injection

Reply:
haha 
Yes, I've already tried and it won't work. Because the SQL query is no longer valid afterwards. As this is how it's done in other repositories, I've used the same one. 
And as I intend to add a strict map of the possible orders in the upper layers code, this will add security at this time 🤷‍♂️